www/js/components/pages/vm/disks.js

```diff
@@ -10,43 +10,133 @@
  */
 
 var Backbone = require('backbone');
-var Disks = Backbone.Collection.extend({});
 
+/**
+ * Disks Table/List
+ */
+var Disks = Backbone.Collection.extend({});
+var JobProgressView = require('../../../views/job-progress');
+var DisksFormView = require('./disks-form');
 var DiskRowTemplate = require('./disks-row.hbs');
+
+var adminui = require('../../../adminui');
 var DiskRow = Backbone.Marionette.ItemView.extend({
     tagName: 'tr',
     template: DiskRowTemplate,
     events: {
+        'click .edit': 'resizeDisk',
+        'click .delete': 'deleteDisk'
     },
-    initialize: function (options)  {
-        // Override list initialization
+
+    deleteDisk: function () {
+        if (window.confirm('Are you sure you want to delete this disk?')) {
+            this.vm.removeDisk(this.model.get('pci_slot'),
+                this._onJob.bind(this));
+        }
+    },
+    _onJob: function (err, job) {
+        if (err) {
+            console.log('[job error]: ', err);
+            var msg = (err.responseText) ?
+                    JSON.parse(err.responseText).message:
+                    'Error creating job';
+            window.alert(msg);
```

Comment:
Should this use a notification instead (as I see some places use notification for job failures):
```
                adminui.vent.trigger('notification', {
                    level:'error',
                    message: msg
                });
```

Reply:
Cannot see why not. Changed!